test_suite/generic/test_generic.py

```diff
@@ -877,11 +897,6 @@ def test_kdump_status(self, host):
         """
         with host.sudo():
             kernel_version = host.check_output('uname -r').split("-")[0]
-            arch = host.system_info.arch
```

Comment:
Do you mean to remove the fixture and add all this logic inside the test case?
I slightly disagree cause fixtures help keep test code modular and organized by separating setup code from test logic, the tests become cleaner and easier to understand and maintain.
If you notice I use this fixture for two test cases and not only for one, and who knows maybe we will find out this error will appear on another test case in the future so it is also reusable.
In addition, this fixture won't even start that test if the condition does not apply cause it will skip immediately.

Reply:
Thanks for the explanation. 
Indeed, I agree it's preferable for tests not to even run whenever possible :+1: